PocketKit/Sources/PocketKit/Home/HomeCarouselItemCell.swift

```diff
@@ -0,0 +1,285 @@
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension HomeCarouselItemCell {
+    struct Model: Hashable {
+        let title: String?
+        let domain: String?
+        let timeToRead: String?
+        let thumbnailURL: URL?
+        let saveButtonMode: RecommendationSaveButton.Mode?
+        
+        init(item: ItemsListItem) {
+            self.title = item.title
+            self.domain = item.domainMetadata?.name ?? item.domain ?? item.bestURL?.host
+            let timeToRead = item.timeToRead ?? 0
+            self.timeToRead = timeToRead > 0 ? "\(timeToRead) min read" : nil
+            self.thumbnailURL = imageCacheURL(for: item.topImageURL)
+            self.saveButtonMode = nil
+        }
+        
+        init(viewModel: HomeRecommendationCellViewModel) {
+            self.title = viewModel.title
+            self.domain = viewModel.domain
+            self.timeToRead = viewModel.timeToRead
+            self.thumbnailURL = viewModel.imageURL
+            self.saveButtonMode = viewModel.saveButtonMode
+        }
+
+        var favoriteAction: ItemAction? = nil
+        var overflowActions: [ItemAction]? = nil
+        
+        var attributedTitle: NSAttributedString {
+            NSAttributedString(string: title ?? "", style: .title)
+        }
+        
+        var attributedDomain: NSAttributedString {
+            return NSAttributedString(string: domain ?? "", style: .domain)
+        }
+        
+        var attributedTimeToRead: NSAttributedString {
+            return NSAttributedString(string: timeToRead ?? "", style: .timeToRead)
+        }
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        model = nil
+    }
+
+    func reconfigure() {
+        titleLabel.attributedText = model?.attributedTitle
+        domainLabel.attributedText = model?.attributedDomain
+        timeToReadLabel.attributedText = model?.attributedTimeToRead
+        
+        if model?.attributedTimeToRead.string == "" {
+            timeToReadLabel.isHidden = true
+        } else {
+            timeToReadLabel.isHidden = false
+        }
+        
+        favoriteButton.accessibilityLabel = model?.favoriteAction?.title
+        favoriteButton.accessibilityIdentifier = model?.favoriteAction?.accessibilityIdentifier
+        favoriteButton.configuration?.image = model?.favoriteAction?.image?.resized(to: Constants.actionButtonImageSize)
+
+        if let favoriteAction = UIAction(model?.favoriteAction) {
+            favoriteButton.addAction(favoriteAction, for: .primaryActionTriggered)
+        }
+        
+        if let mode = model?.saveButtonMode {
+            saveButton.isHidden = false
+            saveButton.mode = mode
+        } else {
+            saveButton.isHidden = true
+        }
+
+        let menuActions = model?.overflowActions?.compactMap(UIAction.init) ?? []
+        overflowButton.menu = UIMenu(children: menuActions)
```

Comment:
Let's update the recommendation cell (hero) to follow this same pattern of setting the overflow button's menu within the view (as opposed to doing so in the view controller)